### ISSUE TITLE
asciidoctor broken on darwin when installed using nix-actions

### ISSUE THREAD

**Issue by nomeata** [cachix/install-nix-action#22]
Not sure where the problem is, but I’ll start here; maybe it’s actually a problem with nix or nixpkgs or darwin or Github Actions…

Consider the repo https://github.com/nomeata/asciidoctor-nix-darwin-bug. It builds something with asciidoctor, a ruby program.

It fails when run on drawin via Github Actions and the nix install action.

It succeeds on Drawin elsewhere, or on Github actions on Linux.

Maybe it is related to the use of `/opt/nix`?

See https://github.com/nomeata/asciidoctor-nix-darwin-bug/runs/461231390?check_suite_focus=true for a build log of this:
```
Traceback (most recent call last):
        12: from /nix/store/6z2m80sza5accbbyxisp0ffwqphs1sxr-asciidoctor-2.0.10/bin/asciidoctor:16:in `<main>'
        11: from /nix/store/1cx5njhpnj9mk6vviijph69f49dldw2m-bundler-1.17.3/lib/ruby/gems/2.6.0/gems/bundler-1.17.3/lib/bundler.rb:101:in `setup'
        10: from /nix/store/1cx5njhpnj9mk6vviijph69f49dldw2m-bundler-1.17.3/lib/ruby/gems/2.6.0/gems/bundler-1.17.3/lib/bundler.rb:134:in `definition'
         9: from /nix/store/1cx5njhpnj9mk6vviijph69f49dldw2m-bundler-1.17.3/lib/ruby/gems/2.6.0/gems/bundler-1.17.3/lib/bundler.rb:66:in `configure'
         8: from /nix/store/1cx5njhpnj9mk6vviijph69f49dldw2m-bundler-1.17.3/lib/ruby/gems/2.6.0/gems/bundler-1.17.3/lib/bundler.rb:534:in `configure_gem_home_and_path'
         7: from /nix/store/1cx5njhpnj9mk6vviijph69f49dldw2m-bundler-1.17.3/lib/ruby/gems/2.6.0/gems/bundler-1.17.3/lib/bundler.rb:541:in `configure_gem_path'
         6: from /nix/store/1cx5njhpnj9mk6vviijph69f49dldw2m-bundler-1.17.3/lib/ruby/gems/2.6.0/gems/bundler-1.17.3/lib/bundler.rb:351:in `use_system_gems?'
         5: from /nix/store/1cx5njhpnj9mk6vviijph69f49dldw2m-bundler-1.17.3/lib/ruby/gems/2.6.0/gems/bundler-1.17.3/lib/bundler.rb:84:in `configured_bundle_path'
         4: from /nix/store/1cx5njhpnj9mk6vviijph69f49dldw2m-bundler-1.17.3/lib/ruby/gems/2.6.0/gems/bundler-1.17.3/lib/bundler.rb:273:in `settings'
         3: from /opt/nix/store/hkz1a3rg9jbnb6agfqz4bwwjjy5jrpdf-ruby-2.6.5/lib/ruby/2.6.0/rubygems/core_ext/kernel_require.rb:72:in `require'
         2: from /opt/nix/store/hkz1a3rg9jbnb6agfqz4bwwjjy5jrpdf-ruby-2.6.5/lib/ruby/2.6.0/rubygems/core_ext/kernel_require.rb:72:in `require'
         1: from /nix/store/1cx5njhpnj9mk6vviijph69f49dldw2m-bundler-1.17.3/lib/ruby/gems/2.6.0/gems/bundler-1.17.3/lib/bundler/settings.rb:5:in `<top (required)>'
/nix/store/1cx5njhpnj9mk6vviijph69f49dldw2m-bundler-1.17.3/lib/ruby/gems/2.6.0/gems/bundler-1.17.3/lib/bundler/settings.rb:6:in `<module:Bundler>': uninitialized constant Bundler::Settings (NameError)
builder for '/nix/store/cyb5hi1908rd7rp28dd4ndwq16ca5a9p-doc.drv' failed with exit code 1
error: build of '/nix/store/cyb5hi1908rd7rp28dd4ndwq16ca5a9p-doc.drv' failed
```

**Comment by domenkozar** [comment]
Duplicates #21 :)

**Comment by nomeata** [comment]
Not a perfect duplicate, as #21 seems to be about using build products that were built with `/opt/nix` on other machines, where I just want to build on GitHub Actions (but not use it elsewhere). But I guess a fix for #21 might fix the present bug as well, so I’ll wait and watcht that space. Thanks!
